UIforETW/Utility.h

```diff
@@ -76,30 +88,24 @@ int64_t GetFileSize(const std::wstring& path);
 void SetClipboardText(const std::wstring& text);
 std::wstring GetClipboardText();
 
-enum WindowsVersion
-{
-	kWindowsVersionXP,
-	kWindowsVersionVista,
-	kWindowsVersion7,
-	kWindowsVersion8,
-	kWindowsVersion8_1,
-	kWindowsVersion10,
-};
+std::wstring GetEnvironmentVariableString(_In_z_ PCWSTR variable);
+std::string GetEnvironmentVariableString(_In_z_ PCSTR variable);

```

Comment:
Ah - I somehow missed that one when I was looking to see if a narrow character version was ever needed.

If there aren't any added complexities in doing that then that would be the right way to go. That would also get rid of a call to AnsiToUnicode().

However there may be a long tail of other changes needed - systemDrive_, for instance, is a string. However I think that's basically it for std::string, and almost it for AnsiToUnicode.